processor/awscontainerinsightprocessor/processor.go

```diff
@@ -0,0 +1,121 @@
+	stores                  []stores.K8sStore
+	storeRefreshMinInterval time.Duration
+	storeRefreshTimeout     time.Duration
+	storeRefreshStoppedC    chan bool //used only by test
+	shutdownC               chan bool
+	TagService              bool   `toml:"tag_service"`
+	ClusterName             string `toml:"cluster_name"`
+	HostIP                  string `toml:"host_ip"`
+	NodeName                string `toml:"node_name"`
+	PrefFullPodName         bool   `toml:"prefer_full_pod_name"`
+}
+
+// ProcessMetrics takes pdata.Metrics and does metric decorations (e.g. adding resource attributes/new metrics)
+func (acip *awscontainerinsightprocessor) ProcessMetrics(_ context.Context, md pdata.Metrics) (pdata.Metrics, error) {
+	//TODO: add logic to do the metric decoration by adding new resource attributes and datapoints
+	return md, nil
+}
+
+// Shutdown stops the aws container insights processor
+func (acip *awscontainerinsightprocessor) Shutdown() {
+	close(acip.shutdownC)
+}
+
+func (acip *awscontainerinsightprocessor) start() error {
+	acip.NodeName = os.Getenv("HOST_NAME")
+	if acip.NodeName == "" {
+		return errors.New("missing environment variable HOST_NAME. Please check your deployment YAML file")
+	}
+
+	acip.HostIP = os.Getenv("HOST_IP")
+	if acip.HostIP == "" {
+		return errors.New("missing environment variable HOST_IP. Please check your deployment YAML file")
+	}
+
+	acip.shutdownC = make(chan bool)
+
+	// TODO: add the relevant store to acip.stores
+	ctx, cancel := context.WithTimeout(context.Background(), acip.storeRefreshTimeout)
+	go func(ctx context.Context, cancel func()) {
+		for _, store := range acip.stores {
+			store.RefreshTick(ctx)
+		}
+		cancel()
+	}(ctx, cancel)
+	<-ctx.Done()
+	cancel()
+
+	go func() {
+		refreshTicker := time.NewTicker(acip.storeRefreshMinInterval)
+		for {
+			select {
+			case <-refreshTicker.C:
+				ctx, cancel := context.WithTimeout(context.Background(), acip.storeRefreshTimeout)
+				// spawn a goroutine to process the stores
+				go func(ctx context.Context, cancel func()) {
+					for _, store := range acip.stores {
+						store.RefreshTick(ctx)
+					}
+					cancel()
+				}(ctx, cancel)
+				// block until either the goroutine has processed all stores or the timeout expires
+				<-ctx.Done()
+				cancel()
+			case <-acip.shutdownC:
+				//for testing only
+				if acip.storeRefreshStoppedC != nil {
+					close(acip.storeRefreshStoppedC)
+				}
+				refreshTicker.Stop()
+				return
+			}
+		}
+	}()
+
+	return nil
+}
+
+func newAwsContainerInsightProcessor(logger *zap.Logger, config *Config) *awscontainerinsightprocessor {
+	processor := &awscontainerinsightprocessor{
+		logger:                  logger,
+		TagService:              config.TagService,
+		PrefFullPodName:         config.PrefFullPodName,
+		storeRefreshMinInterval: time.Second,
+		storeRefreshTimeout:     30 * time.Second, //make it big enough so that all stores can finish refreshing
+	}
+	if err := processor.start(); err != nil {
```

Comment:
Should the processor actually be started here, or just initialized for something else to start as appropriate?  Also, Shouldn't the start method be exported to satisfy the metrics processor interface?